util/test/sub_test.py

```diff
@@ -740,1716 +741,1756 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
+    chpl_home=os.path.normpath(chpl_home)
+
+    # Find the test directory
+    testdir=chpl_home+'/test'
+    if os.path.isdir(testdir)==0:
+        testdir=chpl_home+'/examples'
+        if os.path.isdir(testdir)==0:
+            Fatal('Cannot find test directory '+chpl_home+'/test or '+testdir)
+    # Needed for MacOS mount points
+    testdir = os.path.realpath(testdir)
+    # sys.stdout.write('testdir='+testdir+'\n');
+
+    # If user specified a different test directory (e.g. with --test-root flag on
+    # start_test), use it instead.
+    test_root_dir = os.environ.get('CHPL_TEST_ROOT_DIR')
+    if test_root_dir is not None:
+        testdir = test_root_dir
+
+    # Get the current directory (normalize for MacOS case-sort-of-sensitivity)
+    localdir = os.path.normpath(os.getcwd()).replace(testdir, '.')
+    # sys.stdout.write('localdir=%s\n'%(localdir))
+
+    if localdir.find('./') == 0:
+        # strip off the leading './'
+        localdir = localdir.lstrip('.')
+        localdir = localdir.lstrip('/')
+    # sys.stdout.write('localdir=%s\n'%(localdir))
+
+    return localdir
+
+def get_chpl_base(compiler):
+    """Returns the normalized path to the base installation."""
+    if not os.access(compiler,os.R_OK|os.X_OK):
+        Fatal('Cannot execute compiler \''+compiler+'\'')
+
+    path_to_compiler=os.path.abspath(os.path.dirname(compiler))
+    # Assume chpl binary is 2 directory levels down in the base installation
+    (chpl_base, tmp) = os.path.split(path_to_compiler)
+    (chpl_base, tmp) = os.path.split(chpl_base)
+    chpl_base=os.path.normpath(chpl_base)
+    # sys.stdout.write('CHPL_BASE='+chpl_base+'\n')
+    return chpl_base
+
+def get_chpl_home(chpl_base):
+    """Returns the normalized path to the Chapel installation."""
+    # If $CHPL_HOME is not set, use the base installation of the compiler
+    chpl_home=os.getenv('CHPL_HOME', chpl_base);
+    chpl_home=os.path.normpath(chpl_home)
+    # sys.stdout.write('CHPL_HOME='+chpl_home+'\n');
+    return chpl_home
+
+def main(): 
+    # Start of sub_test proper
+    #
+    global utildir, chpl_base, chpl_home, machine, envCompopts, platform
+    global chplcommstr, chplnastr, chpllmstr, chpltasksstr
+
+    if len(sys.argv)!=2:
+        print('usage: sub_test COMPILER')
+        sys.exit(0)
+
+    compiler=sys.argv[1]
+    is_chpldoc = compiler.endswith('chpldoc')
+    chpl_base = get_chpl_base(compiler)
+    chpl_home = get_chpl_home(chpl_base)
+    localdir = get_local_dir(chpl_base)
+
+    # Find the test util directory -- set this in start_test to permit
+    # a version of start_test other than the one in CHPL_HOME to be used
+    utildir=os.getenv('CHPL_TEST_UTIL_DIR');
+    if utildir is None or not os.path.isdir(utildir):
+        Fatal('Cannot find test util directory {0}'.format(utildir))
+
+    # Needed for MacOS mount points
+    utildir = os.path.realpath(utildir)
+    # sys.stdout.write('utildir='+utildir+'\n');
+
```

Comment:
Could this also be factored into `get_utildir`?

Reply:
Done